app/controllers/application_controller.rb

```diff
@@ -41,7 +35,17 @@ def patron_info_response
     end
 
     def authenticate_user!
-      redirect_to root_url unless current_user?
+      if current_user?
+        return renew_session_token if patron.stale?
+      else
+        redirect_to root_url
+      end
```

Comment:
For me this is a little more readable, just adding one line (`renew_session_token if patron.stale?
`) instead of a kind of complex nested if thing:

```suggestion
      redirect_to root_url unless current_user?
      renew_session_token if patron.stale?
```